registry/txt.go

```diff
@@ -201,26 +206,49 @@ type nameMapper interface {
 	toTXTName(string) string
 }
 
-type prefixNameMapper struct {
+type affixNameMapper struct {
 	prefix string
+	suffix string
 }
 
-var _ nameMapper = prefixNameMapper{}
+var _ nameMapper = affixNameMapper{}
 
-func newPrefixNameMapper(prefix string) prefixNameMapper {
-	return prefixNameMapper{prefix: strings.ToLower(prefix)}
+func newaffixNameMapper(prefix string, suffix string) affixNameMapper {
+	return affixNameMapper{prefix: strings.ToLower(prefix), suffix: strings.ToLower(suffix)}
 }
 
-func (pr prefixNameMapper) toEndpointName(txtDNSName string) string {
+/* func (pr affixNameMapper) toEndpointName(txtDNSName string) string {
 	lowerDNSName := strings.ToLower(txtDNSName)
-	if strings.HasPrefix(lowerDNSName, pr.prefix) {
+	regex := regexp.MustCompile(`\.`)
+    DNSName := regex.Split(lowerDNSName, 2)
+	if (strings.HasPrefix(DNSName[0], pr.prefix) && strings.HasSuffix(DNSName[0], pr.suffix)) {
+		DNSName[0] = strings.TrimPrefix(DNSName[0], pr.prefix)
+		DNSName[0] = strings.TrimSuffix(DNSName[0], pr.suffix)
+		return DNSName[0] + "." + DNSName[1]
+	}
+	return ""
+}*/
```

Comment:
I think the commented code should be removed.

Reply:
You're right. Missed this one after testing.

Reply:
@ruudk Done